php/php.editor/src/org/netbeans/modules/php/editor/parser/astnodes/StaticConstantAccess.java

```diff
@@ -54,10 +62,24 @@ public Identifier getConstantName() {
         while (expression instanceof ExpressionArrayAccess) {
             expression = ((ExpressionArrayAccess) expression).getExpression();
         }
+        // can use dynamic constant names since PHP 8.3
+        // e.g. Foo::{$bar};
+        assert expression != null;
+        if (isDynamicName) {
+            // XXX create an identifier to avoid NPE, can improve this?
```

Comment:
`XXX` - do we want to keep ii in sources?

Reply:
Will remove `XXX`. Thanks!